### PR TITLE
Add status popover component DAH-158

### PR DESCRIPTION
**Pull request by akegan**
DAH-158

This PR:
1. Adds a popover component that tracks a given input button
1. Adds a sample implementation with scrollable status items that we can mimic for the final usage in PopoverWrapper

Review instructions in the ticket

### CODE REVIEW

**Comment by akegan** [comment]
working on moving the popover css into its own file now

File: app/javascript/components/molecules/Popover.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+// Source: https://www.30secondsofcode.org/react/s/use-click-outside
+const useClickOutside = (ref, callback) => {
+  const handleClick = (e) => {
+    if (ref.current && !ref.current.contains(e.target)) {
+      callback()
+    }
+  }
+
+  React.useEffect(() => {
```
Comment (james-wills-sf) [inline app/javascript/components/molecules/Popover.js]: nit: you can import useEffect directly

File: app/javascript/components/molecules/Popover.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+          offset: [0, 32]
+        }
+      }
+    ]
+  })
+
+  return (
+    <>
+      {buttonElement({ ref: buttonRef, onClick: () => setShowPopper(!showPopper) })}
+      {showPopper ? (
```
Comment (james-wills-sf) [inline app/javascript/components/molecules/Popover.js]: you can just do `showPopper && ...` instead of `showPopper ? ... : null`

**Review by james-wills-sf — APPROVED** [review]
@akegan QA lgtm!

**Pull request merged**